### ISSUE TITLE
Cannot set config option with struct type from command-line

### ISSUE THREAD

**Issue by LordAro** [astral-sh/ruff#13995]
While trying out some adhoc testing, I ran the following:
```
ruff check --extend-select=PT --config 'lint.flake8-pytest-style="csv"'
```

which resulted in the error:

```
error: invalid value 'lint.flake8-pytest-style="csv"' for '--config <CONFIG_OPTION>'

  tip: A `--config` flag must either be a path to a `.toml` configuration file
       or a TOML `<KEY> = <VALUE>` pair overriding a specific configuration
       option

Could not parse the supplied argument as a `ruff.toml` configuration option:

invalid type: string "csv", expected struct Flake8PytestStyleOptions
in `lint`

For more information, try '--help'
```

according to the tip (and the help text) it *should* work, but... doesn't. I'm not sure if there is a format where it does work, but if nothing else it's unclear.

I can imagine this applies to other options that use this form, but this is the only one I've found.

Ruff: 0.7.1

**Comment by MichaReiser** [comment]
I think  you're missing the last name of the setting: `lint` and `flake8-pytest-style` are only the setting group:

```
ruff check --extend-select=PT --config lint.flake8-pytest-style.parametrize-names-type="csv"
```

**Comment by MichaReiser** [comment]
Okay, it's slightly more complicated because the value must be a string but this should work

```bash
ruff check --extend-select=PT --config "lint.flake8-pytest-style.parametrize-names-type='csv'"
```

**Comment by LordAro** [comment]
Oh you're quite right, I'm a fool and managed to only copy the section name rather than the rule itself.

Not that I want to be that guy, but maybe the error message could be improved somehow? Mentioning a struct seems unhelpful...

**Comment by MichaReiser** [comment]
> Not that I want to be that guy, but maybe the error message could be improved somehow? Mentioning a struct seems unhelpful...

That makes a lot of sense to me. I also had to play with the CLI to determine the right syntax (see my first answer). I'm not sure how much we can do about the struct part without writing our own deserialization framework but we should take a look if there are any low hanging improvements
